implement upload logo feature

RyanCoulsonCA opened the pull request:
Closes #68 

Allows the user to upload a logo image to the metadata page. The uploaded image is previewed to the user. When the user saves the metadata (at the moment this involves pressing the "TESTING CONFIG" button), the image is uploaded to the file directory and is properly set in the configuration file.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ramp4-pcar4/storylines-editor/74)
<!-- Reviewable:end -->

mohsin-r reviewed (approved):
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/storylines-editor/74)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @RyanCoulsonCA)



<!-- Sent from Reviewable.io -->

RyanCoulsonCA commented:
Rebased this PR, and added some additional functionality:

* loading an existing product will still display the logo preview
* added additional logic to make "browse" button appear outside of the input box

RyanCoulsonCA commented:
[Updated Demo Link](https://ramp4-pcar4.github.io/storylines-editor/fix-68/#/en/editor)

yileifeng reviewed (approved):
Loading in existing product does not display logo or URL field.

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/storylines-editor/74#-:-NOFSC__-ji-56qqp03l:bom8wjv)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @RyanCoulsonCA)



<!-- Sent from Reviewable.io -->

RyanCoulsonCA reviewed (commented):
Created a [new issue](https://github.com/ramp4-pcar4/storylines-editor/issues/80) for this. Will work on it as soon as I finish up #78.

> *[Reviewable](https://reviewable.io/reviews/ramp4-pcar4/storylines-editor/74#-:-NOFSc_w2BB3cYq89ryY:b-2w1lia)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @RyanCoulsonCA)



<!-- Sent from Reviewable.io -->
